src/setup.rs

```diff
@@ -0,0 +1,261 @@
+                        println!("{}\n", "  ✔️ Done".green());
+                    }
+                })
+                .expect("Couldn't create new user");
+
+            println!("{}\n", "About to create a new PostgreSQL table named 'plume'".blue());
+            Command::new("createdb")
+                .arg("-O")
+                .arg("plume")
+                .arg("plume")
+                .status()
+                .map(|s| {
+                    if s.success() {
+                        println!("{}\n", "  ✔️ Done".green());
+                    }
+                })
+                .expect("Couldn't create new table");
+
+            migrate();
+
+            init_pool()
+                .expect("Couldn't init DB pool")
+                .get()
+                .map(|c| DbConn(c))
+                .expect("Couldn't connect to the database")
+        }
+    }
+}
+
+fn migrate() {
+    println!("{}\n", "Running migrations…".blue());
+    Command::new("diesel")
+        .arg("migration")
+        .arg("run")
+        .arg("--database-url")
+        .arg(DB_URL.as_str())
+        .status()
+        .map(|s| {
+            if s.success() {
+                println!("{}\n", "  ✔️ Done".green());
+            }
+        })
+        .expect("Couldn't run migrations");
+}
+
+fn setup_type(conn: DbConn) {
+    println!("\nDo you prefer a simple setup, or to customize everything?\n");
+    println!("  1 - Simple setup");
+    println!("  2 - Complete setup");
+    match read_line().as_ref() {
+        "Simple" | "simple" | "s" | "S" |
+        "1" => quick_setup(conn),
+        "Complete" | "complete" | "c" | "C" |
+        "2" => complete_setup(conn),
+        x => {
+            println!("Invalid choice. Choose between '1' or '2'. {}", x);
+            setup_type(conn);
+        }
+    }
+}
+
+fn quick_setup(conn: DbConn) {
+    println!("What is your instance domain?");
+    let domain = read_line();
+    write_to_dotenv("BASE_URL", domain.clone());
+
+    println!("\nWhat is your instance name?");
+    let name = read_line();
+
+    let instance = Instance::insert(&*conn, NewInstance {
+        public_domain: domain,
+        name: name,
+        local: true
+    });
+
+    println!("{}\n", "  ✔️ Your instance was succesfully created!".green());
+
+    // Generate Rocket secret key.
+    let key = Command::new("openssl")
+        .arg("rand")
+        .arg("-base64")
+        .arg("32")
+        .output()
+        .map(|o| String::from_utf8(o.stdout).expect("Invalid output from openssl"))
+        .expect("Couldn't generate secret key.");
+    write_to_dotenv("ROCKET_SECRET_KEY", key);    
+
+    create_admin(instance, conn);
+}
+
+fn complete_setup(conn: DbConn) {
+    quick_setup(conn);
+
+    println!("\nOn which port should Plume listen? (default: 7878)");
+    let port = read_line_or("7878");
+    write_to_dotenv("ROCKET_PORT", port);
+
+    println!("\nOn which port should Plume listen? (default: 0.0.0.0)");
```

Comment:
Shouldn't it be on which ip?

Reply:
(oops, forgot to change that after having copy/pasted the previous lines)